src/movies.js

```diff
@@ -1,16 +1,108 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+function getAllDirectors(movies) {
+    let directors = movies.map(movie => {
+          return movie.director;
+      });
+    return directors;
+  }
+  getAllDirectors(movies);
+
+// _Bonus_: It seems some of the directors had directed multiple 
+// movies so they will pop up multiple times in the array of directors. 
+// How could you "clean" a bit this array and make it unified (without duplicates)?
+
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+// function howManyMovies(movies) {
+    function howManyMovies(movies) {
+        let numberOfMovies = movies.filter(movie => {
+        return movie.director === 'Steven Spielberg' && movie.genre.indexOf('Drama') !== -1;
+      }).length;
+    return numberOfMovies
+    }
+    howManyMovies(movies);
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+//summed value / number of elements
+function ratesAverage(movies) {
+    let sum = movies.reduce((total, rates) => {
+      return total + rates.rate; 
+    }, 0)
+    let average = sum / 250;
+    return Number(average.toFixed(2))
+}
```

Comment:
here you needed to add an if statement to check if the movies array is empty so then you should return 0.
instead of hard coding to get the average like = sum / 250 you could have used the length of the movies array.
and to sum only the movie rate of the ones that have a rate you could add the if statement inside of the reduce method.
```
function ratesAverage(movies) {
  if (movies.length === 0) return 0;
  let sum = movies.reduce((total, rates) => {
    if (rates.rate) {
      return total + rates.rate;
    } else {
      return total;
    }
  }, 0);
  let average = sum / movies.length;
  return Number(average.toFixed(2));
}
```